xarray/tests/test_dataarray.py

```diff
@@ -913,6 +913,88 @@ def test_swap_dims(self):
         actual = array.swap_dims({'x': 'y'})
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_expand_dims(self):
+        array = DataArray(np.random.randn(3, 4), dims=['x', 'dim_0'],
+                          coords={'x': np.linspace(0.0, 1.0, 3)},
+                          attrs={'key': 'entry'})
+
+        with self.assertRaises(ValueError):
+            array.expand_dims(0)  # the first argument should not be an integer
+        with self.assertRaises(ValueError):
+            # dims and axis argument should be the same length
+            array.expand_dims(dim=['a', 'b'], axis=[1, 2, 3])
+        with self.assertRaises(ValueError):
+            # Should not pass the already existing dimension.
+            array.expand_dims(dim=['x'])
+        # raise if duplicate
+        with self.assertRaises(ValueError):
+            array.expand_dims(dim=['y', 'y'])
+        with self.assertRaises(ValueError):
+            array.expand_dims(dim=['y', 'z'], axis=[1, 1])
+        with self.assertRaises(ValueError):
+            array.expand_dims(dim=['y', 'z'], axis=[2, -2])
+
+        # Working test
```

Comment:
Consider breaking these into smaller test methods, e.g., `test_expand_dims_errors`. It's  easier to debug when tests each one thing.